java/src/jmri/jmrit/simpleclock/SimpleClockFrame.java

```diff
@@ -561,6 +563,17 @@ private void setTimeSourceChanged() {
         if (index == internalSourceIndex) {
             clock.setInternalMaster(true, true);
         } else {
```

Comment:
This "else" block, I _assume_, is for JMRI acting as other-than-master.  That block is where your functional changes are applied.  So I don't see how these changes _allow_ "JMRI acting as fast clock master" to use non-integer fast rates.  

Based on my assumptions, these code changes seem only to disallow "non-integer fast rates on connections disallowing non-integer fast rates when JMRI is _not_ acting as fast clock master.  (Of course, this fast-clock stuff is very obscure, and my _assumptions_, above, may be entirely wrong...)